src/main/drivers/display.c

```diff
@@ -85,8 +134,43 @@ int displayWrite(displayPort_t *instance, uint8_t x, uint8_t y, const char *s)
 
 int displayWriteWithAttr(displayPort_t *instance, uint8_t x, uint8_t y, const char *s, textAttributes_t attr)
 {
-    instance->posX = x + strlen(s);
+    size_t length = strlen(s);
+
+    instance->posX = x + length;
     instance->posY = y;
+
+    if (displayAttributesRequireEmulation(instance, attr)) {
+        // We can't overwrite s, so we use an intermediate buffer if we need
+        // text attribute emulation.
+        char buf[DISPLAY_MAX_STRING_SIZE + 1];
+        textAttributes_t blockAttr;
+        size_t blockSize = length > sizeof(buf) ? sizeof(buf) : length;
+        if (displayEmulateTextAttributes(instance, buf, s, blockSize, attr, &blockAttr)) {
+            // Emulation requires rewriting the string, which might be bigger than our buffer.
+            int ret = instance->vTable->writeString(instance, x, y, buf, blockAttr);
+#ifdef DISPLAY_ATTRIBUTE_EMULATION_MULTIPLE_WRITES
```

Comment:
I'm not sure that `DISPLAY_ATTRIBUTE_EMULATION_MULTIPLE_WRITES` will be ever used.

Reply:
We can delete that, I guess we can always recover it from the history if we ever need it.

Reply:
Ok

Reply:
By deleting it we can also simplify `displayEmulateTextAttributes()` since it can now just overwrite the received attributes rather than copying them, saving 16 bytes. Let's go for it.